Basic Samurai Support added

Rjb99 opened the pull request:
Added sam actions and statuses

used the base cast times for Iaijutsu skills.

NamelessFaceless commented on src/data/ACTIONS/SAM.js:
```diff
@@ -0,0 +1,240 @@
+	HISSATSU_GUREN:
+	{
+		id:7496,
+		name: 'Hisstasu: Guren',
```
just a typo. xd

ackwell replied:
☝️

ackwell commented on src/data/ACTIONS/SAM.js:
```diff
@@ -0,0 +1,240 @@
+	//-----
+	//Player GCDs
+	//-----
+	HAKAZE: 
```
Not sure if `eslint` picks this up, but please follow the convention in the rest of the codebase for this, e.g.:

```js
export default {
	HAKAZE: { // on same line
		key: value, // space between key and val
	},
}
```

ackwell commented on src/data/STATUSES/SAM.js:
```diff
@@ -0,0 +1,39 @@
+export default {
+	
+	THIRD_EYE: 
```
Same deal as actions file

ackwell commented on src/data/ACTIONS/SAM.js:
```diff
@@ -0,0 +1,240 @@
+        	id:7477,
+        	name: 'Hakazae',
+        	icon: 'https://secure.xivdb.com/img/game/003000/003151.png',
+        	onGCD: true,
```
`onGcd`

ackwell commented on src/data/ACTIONS/SAM.js:
```diff
@@ -0,0 +1,240 @@
+		id:7498,
+		name:'Third Eye',
+		icon:'https://secure.xivdb.com/img/game/003000/003153.png'
+		onGCD: false,
```
No need to include `onGcd` when it isn't one

ackwell commented on src/parser/jobs/sam/About.js:
```diff
@@ -0,0 +1,23 @@
+		</Message>
+	</Fragment>
+	supportedPatch = '4.35'
+	contributors = 
```
Looks like something happened w/ indentation here.

ackwell commented on src/parser/jobs/sam/About.js:
```diff
@@ -0,0 +1,23 @@
+import CoreAbout from 'parser/core/modules/About'
+
+export default class About extends CoreAbout {
+	description = <Fragment>
```
I won't block on this, but see the pins in `#development` re: descriptions.

ackwell commented on src/parser/jobs/sam/Higanbana.js:
```diff
@@ -0,0 +1,105 @@
+import {Rule, Requirement} from 'parser/core/modules/Checklist'
+import {Suggestion, SEVERITY} from 'parser/core/modules/Suggestions'
+
+const STATUS_DURATION = 
```
Indentation

Rjb99 commented:
changed the code to match the style changes requested and fixed all the onGCDs

added bare bones samurai support:
-higanbana up time is tracked
-kaiten usage on GCDs that are not sen moves is also tracked

also updated several wrong ids (whoops) and added statuses for the dot and kaiten buff.
